components/list/list.js

```diff
@@ -91,17 +96,36 @@ class List extends SelectionMixin(LitElement) {
 		`;
 	}
 
+	getItems() {
+		const slot = this.shadowRoot.querySelector('slot:not([name])');
+		if (!slot) return [];
+		return slot.assignedNodes({ flatten: true }).filter((node) => {
+			return node.nodeType === Node.ELEMENT_NODE && (node.role === 'rowgroup' || node.role === 'listitem');
+		});
+	}
+
+	getListItemByKey(key) {
+		const items = this.getItems();
+		for (let i = 0; i < items.length; i++) {
```

Comment:
Nit: could use `Array`'s `find()` method.

Reply:
I think the issue with `find` in this case is that the delegate returns `true/false`, not the item itself.  So I think it would work for a flat list, but not a nested list.  In the nested list case, it would end up returning the root list element that is the ancestor of the nested list item.

Reply:
Ah, gotcha!